### PR TITLE
Adding hook for metronome regen

### PR DESCRIPTION
**Pull request by yalh76**
## Problem
- *In case of `yunohost tools regen-conf --force`, metronome specific config is trashed*

## Solution
- *Adding hook for metronome regen*

## PR Status
- [ ] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

## Package_check results
---
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/jitsi_ynh%20PR11%20(yalh76)/)

### CODE REVIEW

**Comment by yalh76** [comment]
> Dunno if that's really a clear explanation because that's tricky to explain but if you're around in the brique camp during the week I can tell you more about it ;P

Thanks for the informations, I will look at it. I hope to be at the brique camp on Wednesday night and maybe Thursday

**Comment by yalh76** [comment]
> Uuuuh I think you should consider having a look at the yunohost conf regen hook to see how they work.
> 
> so for instance here : https://github.com/YunoHost/yunohost/blob/stretch-unstable/data/hooks/conf_regen/12-metronome#L60-L74
> 
> You should feed `$4` (the 'pending conf dir') to `do_pre_regen` and use it instead of defining your own temp_dir.
> 
> In fact ... the metronome hook runs _before_ your hooks and the whole thing should looks something like :
> 
>     1. core Metronome pre_regen, puts file in `${pending_dir}/etc/metronome`
> 
>     2. your app pre_regen, should probably edit/patch the conf already in `${pending_dir}/../metronome/etc/metronome` (or maybe add additional files, idk your use case precisely).
>        
>        * for an example, check mailman [here](https://github.com/YunoHost-Apps/mailman_ynh/blob/master/sources/hooks/conf_regen/98-postfix_mailman#L9-L16)
> 
>     3. ...yunohost decide or not to actually apply the conf from ${pending_dir} or not, c.f. the --dry-run option where the conf is generated but not actually applied in the "true" directories...
> 
>     4. core post_regen (where it reloads the service possibly or fix some permission and other weird stuff _but it has nothing to do anymore about writing conf files_ anymore !)
> 
>     5. your app post_regen ... i'm quite sure that you have nothing to do in it

@alexAubin 

I have completely review the hook.

1. Generating templates of the metronome config during install
2. The hook copy the metronome template in the pending (I wasn't able to use ${pending_dir}/etc/metronome, $pending seems to not have a value)
3. Tested and working.

I you can have a little look before merging

File: scripts/install [hunk 1]
```diff
@@ -139,22 +140,26 @@ fi
...
-ynh_replace_string --match_string="__DOMAIN__" --replace_string="$domain" --target_file="$metronome"
-ynh_replace_string --match_string="__APP__" --replace_string="$app" --target_file="$metronome"
-ynh_replace_string --match_string="__VIDEOBRIDGE_SECRET__" --replace_string="$videobridge_secret" --target_file="$metronome"
-ynh_replace_string --match_string="__FOCUS_SECRET__" --replace_string="$focus_secret" --target_file="$metronome"
-ynh_replace_string --match_string="__PORT_COMPONENT__" --replace_string="$port_component" --target_file="$metronome"
-chown -R metronome:metronome "/etc/metronome/conf.d/"
-
+# Add Metronome domain conf template
+metronome_conf="/usr/share/yunohost/templates/jitsi/$domain.cfg.lua"
+mkdir -p /usr/share/yunohost/templates/jitsi/
```
Comment (alexAubin) [inline scripts/install]: Wouldn't it be much simpler to store these files in something like `/etc/yunohost/apps/jistsi/rendered_templates/` ? That way you don't have to worry about remove (the whole folder `/etc/yunohost/apps/jitsi` will be removed automatically) and same for backup

File: scripts/install [hunk 2]
```diff
@@ -139,22 +140,26 @@ fi
...
+ynh_replace_string --match_string="__DOMAIN__" --replace_string="$domain" --target_file="$metronome_conf"
+ynh_replace_string --match_string="__APP__" --replace_string="$app" --target_file="$metronome_conf"
+ynh_replace_string --match_string="__VIDEOBRIDGE_SECRET__" --replace_string="$videobridge_secret" --target_file="$metronome_conf"
+ynh_replace_string --match_string="__FOCUS_SECRET__" --replace_string="$focus_secret" --target_file="$metronome_conf"
+ynh_replace_string --match_string="__PORT_COMPONENT__" --replace_string="$port_component" --target_file="$metronome_conf"
+
+touch "/usr/share/yunohost/templates/jitsi/auth.$domain.cfg.lua"
+touch "/usr/share/yunohost/templates/jitsi/conference.$domain.cfg.lua"
+touch "/usr/share/yunohost/templates/jitsi/jitsi-videobridge.$domain.cfg.lua"
+touch "/usr/share/yunohost/templates/jitsi/focus.$domain.cfg.lua"
```
Comment (alexAubin) [inline scripts/install]: Not sure why you touch these o.O ? Is this really what you mean to do ?

File: conf/metronome_regen_conf.hook [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+force=${2:-0}  # 0/1 --force argument
+dryrun=${3:-0}  # 0/1 --dry-run argument
+pending_conf=$4 # Path of the pending conf file
+
+do_pre_regen() {
+  # Add specific domain metronome conf
+  metronome_conf_dir="/home/yunohost.conf/pending/metronome/etc/metronome/conf.d"
```
[the remark this answers is not in the document]
Comment (alexAubin) [reply conf/metronome_regen_conf.hook]: Ah my bad I didnt see your comment ... but then are you sure that you wrote the right name ? (You mentioned `pending_dir`)

File: scripts/install (hunk 2 above)
Comment (yalh76) [reply scripts/install]: the touch of the files is to create empty files that will replace the metronome config files created when adding the differents sub-domains

File: conf/metronome_regen_conf.hook (hunk 3 above)
[the remark this answers is not in the document]
Comment (yalh76) [reply conf/metronome_regen_conf.hook]: Fixed using pending_conf.

File: scripts/install (hunk 1 above)
Comment (yalh76) [reply scripts/install]: I made it this way, thinking the case of multi_instance. 
I needed a place, independent of the app (ex: `jisti`, `jitsi__2`, `jitsi__3`), to store the metronome configuration of each domain and sub-domains.
And an unique place that can be use by the hook...

**Pull request merged**